### PR TITLE
(EAI-602): Refactor preprocessing to support retrieval evals

### PR DESCRIPTION
**Pull request by mongodben**
Jira: https://jira.mongodb.org/browse/EAI-602

## Changes

- [x] Separate `retrieveRelevantContent` module
- [x] Evals for `retrieveRelevantContent`
- [x] Add relevant metadata to user message custom data
- [x] fix link fuzzy match bug

## Notes

-

### CODE REVIEW

File: packages/chatbot-server-mongodb-public/src/processors/makeStepBackRagGenerateUserPrompt.ts [hunk 1]
```diff
@@ -137,8 +136,12 @@ export const makeStepBackRagGenerateUserPrompt = ({
         url: c.url,
         score: c.score,
       })),
-      customData,
-      preprocessedContent: transformedUserQuery,
```
Comment (nlarew) [inline packages/chatbot-server-mongodb-public/src/processors/makeStepBackRagGenerateUserPrompt.ts:141]: To be clear - we will no longer be using the `preprocessedContent` field in user messages? Do you know if anything relies on this field, e.g. in `scripts` or some other package?

File: packages/chatbot-server-mongodb-public/src/processors/extractMongoDbMetadataFromUserMessage.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import {
   mongoDbProductNames,
   mongoDbProgrammingLanguageIds,
 } from "../mongoDbMetadata";
+import { wrapTraced } from "braintrust";
```
Comment (nlarew) [inline packages/chatbot-server-mongodb-public/src/processors/extractMongoDbMetadataFromUserMessage.ts:12]: I think this is an unused import + there's a few more throughout

**Review by nlarew — APPROVED** [review]
LGTM - one question + cleanup unused imports

File: packages/chatbot-server-mongodb-public/src/processors/makeStepBackRagGenerateUserPrompt.ts (hunk 1 above)
Comment (mongodben) [reply packages/chatbot-server-mongodb-public/src/processors/makeStepBackRagGenerateUserPrompt.ts:141]: the only place it exists in the `scripts` package is within `Omit<T, "preprocessedContent"` type defs, so i think safe to remove. no other meaningful usage elsewhere

**Pull request merged**